Fix unused variable warnings

hpvb opened the pull request:
The forth in my quest to make Godot 3.x compile with -Werror on GCC7

hpvb committed:
Fix unused variable warnings

The forth in my quest to make Godot 3.x compile with -Werror on GCC7

akien-mga commented on core/io/resource_format_binary.cpp:
```diff
@@ -854,13 +854,7 @@ void ResourceInteractiveLoaderBinary::open(FileAccess *p_f) {
 
 	bool big_endian = f->get_32();
-#ifdef BIG_ENDIAN_ENABLED
-	endian_swap = !big_endian;
```
That's a member variable, can we really remove this without logic change?

akien-mga commented on core/io/resource_format_binary.cpp:
```diff
@@ -854,13 +854,7 @@ void ResourceInteractiveLoaderBinary::open(FileAccess *p_f) {
-#ifdef BIG_ENDIAN_ENABLED
-	endian_swap = !big_endian;
-#else
-	bool endian_swap = big_endian;
```
That can change the value of `print_bl("endian swap: " + itos(endian_swap));` below. It may have been a bug in the first place to declare this as local, shadowing the member var, but I'd be cautious at any rate.

hpvb replied:
You're right, I didn't remixed l realize that endian_swap was a member, I thought this was just broken. I'll look at this file again.

hpvb replied:
hmm, print_bl() actually compiles to nothing. So this was all a bit pointless:

```c++
//#define print_bl(m_what) print_line(m_what)
#define print_bl(m_what)
```

I think maybe I need to put all of that behind a debug flag and get rid of print_bl.

akien-mga replied:
Well it seems to be a trick to enable some debug output when hacking that code.

hpvb replied:
Yeah, but maybe we have accumulated too many different tricks like this over the years. :)

akien-mga commented on editor/property_editor.cpp:
```diff
@@ -2283,8 +2283,6 @@ void PropertyEditor::_check_reload_status(const String &p_name, TreeItem *item)
 		if (_get_instanced_node_original_property(p_name, vorig) || usage) {
 			Variant v = obj->get(p_name);
-
-			bool has_reload = _is_property_different(v, vorig, usage);
```
Come on :) It's not hard to see that the whole if becomes pointless :)

akien-mga commented on methods.py:
```diff
@@ -1026,9 +1026,6 @@ def build_legacygl_header(filename, include, class_suffix, output_attribs):
             fd.write("\t\t\t{" + x["set_mask"] + "," + x["clear_mask"] + "},\n")
 
         fd.write("\t\t};\n\n")
-    else:
```
At last! Those warnings have been there forever and are spammed at least 200 times during the build :D

akien-mga commented on scene/2d/canvas_item.cpp:
```diff
@@ -337,7 +337,6 @@ void CanvasItem::_update_callback() {
 		notification(NOTIFICATION_DRAW);
 		emit_signal(SceneStringNames::get_singleton()->draw);
 		if (get_script_instance()) {
-			Variant::CallError err;
```
I'm not sure those are actually unused. IIRC I wanted to remove them a year or two ago and was told not to.

akien-mga replied:
Namely https://github.com/godotengine/godot/issues/5568#issuecomment-231007723

hpvb replied:
I'll have a look at the constructor again then, but it looked pretty pointless to me.

hpvb replied:
It has the default constructor and can't do anything. There's no way this can affect anything.

```c++
	struct CallError {
		enum Error {
			CALL_OK,
			CALL_ERROR_INVALID_METHOD,
			CALL_ERROR_INVALID_ARGUMENT,
			CALL_ERROR_TOO_MANY_ARGUMENTS,
			CALL_ERROR_TOO_FEW_ARGUMENTS,
			CALL_ERROR_INSTANCE_IS_NULL,
		};
		Error error;
		int argument;
		Type expected;
	};
```

vnen replied:
I wonder if `call_multilevel()` once returned an error (like `call()` does) and it was changed for some reason leaving behind some `CallError` variables. In any case, there's no error checking code, so this variable is indeed unused.

hpvb commented:
There are still 5 unused variables left. I'll try and fix this before the merge event this evening.

akien-mga commented on servers/visual/visual_server_scene.cpp:
```diff
@@ -2138,14 +2136,19 @@ void VisualServerScene::_bake_gi_probe_light(const GIProbeDataHeader *header, co
-			//print_line("valid cells: " + itos(success_count));
+
+#ifdef DEBUG_ENABLED
+			print_line("BAKE TIME: " + rtos((OS::get_singleton()->get_ticks_usec() - us) / 1000000.0));
```
To be honest I'm not sure if we want to be spammed by this every time using GI probes with a debug build of the editor (which is the default when building from source).

This should either be shown only in verbose more, or more likely commented out again or removed as it's only meant as debugging info for whoever is hacking that code.

hpvb replied:
I'm happy to move it into a more verbose mode, it just seemed like it may be useful for debugging slowness in the actual project. Maybe when using probes in a scene too busy for the hardware.

My thinking is that if these lines are useful for debugging the engine they should just be there rather than require someone to uncomment them. People less familiar with the engine won't know they are there.

akien-mga replied:
I mean they are relevant for developers debugging this class, not engine users. And someone who is editing the file already in the IDE would notice the comments and could uncomment them if they want to see the debug info :)

hpvb replied:
I guess my point is more: how would you even know to look there if you're trying to debug the problems that these print lines were put in for to help solve.

akien-mga replied:
Well you would have an issue with GI Probes, and would check the relevant code to try to debug it, seeing that there are some debug statements just waiting for you to check their return value. Those statements are not meant for normal users who find their light baking is too slow. They will just make a bug report saying "it's slow", and devs will check and debug it.

Those print_line were added by @reduz as he developed the feature, to be able to see what happens. He's done developing it, and kept them commented out for next time he has to dive it in. Either we get rid of them, or keep them commented out.

hpvb replied:
OK, if that's your stance I'll comment them back out, including the timer variable this time.

hpvb commented:
@akien-mga I think I've addressed all of your concerns now, tested platformer 2d and materials test, both importing and playing. Everything seems fine.
